common/persistence/data_manager_interfaces.go

```diff
@@ -1657,7 +1646,8 @@ type (
 
 // IsTimeoutError check whether error is TimeoutError
 func IsTimeoutError(err error) bool {
-	_, ok := err.(*TimeoutError)
+	var timeoutError *TimeoutError
+	ok := errors.As(err, &timeoutError)
```

Comment:
comment: These changes towards using errors as/Is are the right move, but somewhat higher risk than most other similar changes here because there's a risk that, wrongly implemented they'll fail to match and correctly implemented they might subtly change behaviour (making an error suddenly nonretriable). 

This is *all good* and correct, but probably in future worth pulling to a separate PR just due to the different risk profile.